### PR TITLE
Fixed EKSListNodePools's deleted stack status

### LINKED ISSUE #3237: When an EKS node pool CloudFormation stack is deleted manually, it shows up as node pool status `DELETING` on `ListNodePools`
**Describe the bug**

When an EKS node pool CloudFormation stack is deleted manually, it shows up as node pool status `DELETING` on `ListNodePools`.

**Steps to reproduce the issue:**

1. Create an EKS cluster.
2. Delete one of the node pool's CloudFormation stack manually.
3. Issue a `ListNodePools` request.
4. Observe the deleted stack node pool's status to be `DELETING`.

**Expected behavior**

The node pool should show status `DELETED`

**Screenshots**

**Additional context**

This is due to the redesigned node pool listing + status behavior and the reason `DELETE_COMPLETE` stacks currently are transformed into `NodePoolStatusDeleting` node pool status.

### PR DESCRIPTION
**Pull request by pregnor**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | resolves #3237 
| License         | Apache 2.0

### What's in this PR?

Modified the EKS "node pool with deleted stack status" node pool status to `DELETED` from `DELETING`.

### Why?

This state can happen by manually deleting the stack behind Pipeline's back when the listing should show the proper node pool status of being already deleted.

### Additional context

Manual testing involves creating an EKS cluster, deleting a node pool's CF stack manually and issuing a node pool list request.

<details>
  <summary>Manually deleted CF stack</summary>

```shell
> banzai cluster create --file ~/banzai_cluster_create.yaml 

? Do you want a recommendation on your node groups? No
The current state of the request:

{
  "cloud": "amazon",
  "location": "eu-central-1",
  "name": "pregnor-pipeline-3237-2",
  "properties": {
    "eks": {
      "authConfig": {
        "mapAccounts": [
          "account1",
          "otheraccount"
        ],
        "mapRoles": [
          {
            "groups": [
              "system:bootstrappers",
              "system:nodes"
            ],
            "rolearn": "arn:aws:iam::555555555555:role/devel-nodes-NodeInstanceRole-74RF4UBDUKL6",
            "username": "system:node:{{EC2PrivateDNSName}}"
          }
        ],
        "mapUsers": [
          {
            "groups": [
              "system:masters"
            ],
            "userarn": "arn:aws:iam::555555555555:user/admin",
            "username": "admin"
          },
          {
            "groups": [
              "system:masters"
            ],
            "userarn": "arn:aws:iam::111122223333:user/ops-user",
            "username": "ops-user"
          }
        ]
      },
      "iam": {
        "clusterRoleId": "/test/path/elements/pregnor-pipeline-3185-eks-cluster-role",
        "defaultUser": true,
        "nodeInstanceRoleId": "/test/path/elements/pregnor-pipeline-3185-eks-node-instance-role"
      },
      "nodePools": {
        "pool1": {
          "autoscaling": true,
          "count": 1,
          "instanceType": "t2.small",
          "maxCount": 2,
          "minCount": 1,
          "spotPrice": "0.01",
          "subnet": {
            "subnetId": "subnet-0679560513f0899ac"
          },
          "volumeSize": 22
        },
        "pool2": {
          "autoscaling": true,
          "count": 1,
          "instanceType": "t2.small",
          "maxCount": 2,
          "minCount": 1,
          "spotPrice": "0.01",
          "subnet": {
            "subnetId": "subnet-0afa9a1f390f7f7b4"
          },
          "volumeSize": 22
        },
        "pool3": {
          "autoscaling": true,
          "count": 1,
          "instanceType": "t2.small",
          "maxCount": 2,
          "minCount": 1,
          "spotPrice": "0.01",
          "subnet": {
            "subnetId": "subnet-0afa9a1f390f7f7b4"
          },
          "volumeSize": 22
        }
      },
      "subnets": [
        {
          "subnetId": "subnet-0679560513f0899ac"
        },
        {
          "subnetId": "subnet-0afa9a1f390f7f7b4"
        }
      ],
      "version": "1.17.9",
      "vpc": {
        "vpcId": "vpc-0b99b1266f4e6e0c2"
      }
    }
  },
  "secretName": "aws-pregnor-acquia"
}
? Do you want to edit the cluster request in your text editor? No
? Do you want to CREATE the cluster "pregnor-pipeline-3237-2" now? Yes
INFO[0004] cluster is being created                     
INFO[0004] you can check its status with the command `banzai cluster get "pregnor-pipeline-3237-2"` 
Id  Name                   
82  pregnor-pipeline-3237-2

> banzai cluster list

Id  Name                     Distribution  Location      Version  CreatorName  CreatedAt                  Status   StatusMessage     
82  pregnor-pipeline-3237-2  eks           eu-central-1  1.17.9   pregnor      2020-10-17T09:54:52+02:00  RUNNING  Cluster is running

> banzai cluster --cluster-name pregnor-pipeline-3237-2 nodepool list

Name   Size  Autoscaling  MinimumSize  MaximumSize  VolumeSize  InstanceType  Image                  SpotPrice  SubnetID                  Status   StatusMessage
pool1  1     Enabled      1            2            22          t2.small      ami-065ebfccbbb1c5734  0.01       subnet-0679560513f0899ac  READY                 
pool2  1     Enabled      1            2            22          t2.small      ami-065ebfccbbb1c5734  0.01       subnet-0afa9a1f390f7f7b4  READY               
pool3  1     Enabled      1            2            22          t2.small      ami-065ebfccbbb1c5734  0.01       subnet-0afa9a1f390f7f7b4  READY               

# Manually deleted CF stack.

> banzai cluster --cluster-name pregnor-pipeline-3237-2 nodepool list

Name   Size  Autoscaling  MinimumSize  MaximumSize  VolumeSize  InstanceType  Image                  SpotPrice  SubnetID                  Status   StatusMessage
pool1  1     Enabled      1            2            22          t2.small      ami-065ebfccbbb1c5734  0.01       subnet-0679560513f0899ac  READY                 
pool2  1     Enabled      1            2            22          t2.small      ami-065ebfccbbb1c5734  0.01       subnet-0afa9a1f390f7f7b4  DELETED               
pool3  1     Enabled      1            2            22          t2.small      ami-065ebfccbbb1c5734  0.01       subnet-0afa9a1f390f7f7b4  DELETED               
```

</details>

### Checklist

- [x] Implementation tested (with at least one cloud provider)
- [X] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)
- ~OpenAPI and Postman files updated (if needed)~
- ~User guide and development docs updated (if needed)~
- ~Related Helm chart(s) updated (if needed)~

### To Do

- [x] Manual test.

### CODE REVIEW

File: internal/cluster/distribution/eks/service.go [hunk 1]
```diff
@@ -218,7 +222,7 @@ func NewNodePoolStatusFromCFStackStatus(cfStackStatus string) (nodePoolStatus No
 	switch {
 	case strings.HasSuffix(cfStackStatus, "_COMPLETE"):
 		if cfStackStatus == cloudformation.StackStatusDeleteComplete { // Note: CF stack is deleted, but DB entry is still existing.
-			return NodePoolStatusDeleting
+			return NodePoolStatusDeleted
```
Comment (sagikazarmark) [inline internal/cluster/distribution/eks/service.go:225]: Shouldn't this be on its own case?

File: internal/cluster/distribution/eks/service.go [hunk 2]
```diff
@@ -186,6 +186,10 @@ const (
 	// are being provisioned.
 	NodePoolStatusCreating NodePoolStatus = "CREATING"
 
+	// NodePoolStatusDeleted is the status used when the node pool resources
+	// are aleready removed.
```
Comment (sagikazarmark) [inline internal/cluster/distribution/eks/service.go:190]:
```suggestion
	// are already removed.
```

**Review by sagikazarmark — APPROVED** [review]

File: internal/cluster/distribution/eks/service.go (hunk 1 above)
Comment (pregnor) [reply internal/cluster/distribution/eks/service.go:225]: Yeah, it could be.
Gonna update it.

Comment (pregnor) [reply internal/cluster/distribution/eks/service.go:225]: Also the delete in progress is handled as the "in progress" branch's special case.

**Review by pbalogh-sa — APPROVED** [review]

File: internal/cluster/distribution/eks/service.go (hunk 1 above)
Comment (pregnor) [reply internal/cluster/distribution/eks/service.go]: [Fixed](https://github.com/banzaicloud/pipeline/compare/ad95732ec9bff8610d98fb53d12908889923eecd..5c1d744e7fc86f9e1613cc1df3342d345d33ca4a), flattened the switch.

File: internal/cluster/distribution/eks/service.go (hunk 2 above)
Comment (pregnor) [reply internal/cluster/distribution/eks/service.go]: [Fixed](https://github.com/banzaicloud/pipeline/compare/5c1d744e7fc86f9e1613cc1df3342d345d33ca4a..564220550d148e21191fff10185a77b4f446caf9).

The funny thing is, I'm using a spell checker and it didn't go off, checked my custom dictionary and I don't have this word in it, so I checked the internet, but couldn't find any definition, weird.

**Pull request merged**